### ISSUE TITLE
Getting Exception while declaring a queue using (SimpleAmqpClient)

### ISSUE THREAD

**Issue by vikaschamp** [alanxz/SimpleAmqpClient#173]
Hello , 
I am using SimpleAmqpClient in visual studio 2017.
I have implemented it and able tcreate connection in rabbitmq. But while declaring queue and exchange i am getting exception.

Here is my code 

```cpp
#include "stdafx.h"
#define BOOST_NO_CXX11_RVALUE_REFERENCES
#include <SimpleAmqpClient/SimpleAmqpClient.h>
#include <SimpleAmqpClient\Channel.h>


int main()
{
	AmqpClient::Channel::ptr_t connection = AmqpClient::Channel::Create("localhost" ,5672 ,"finsoltech" , "pass123!" , "/");
	
	connection->DeclareQueue("myqueue");
	connection->DeleteQueue("myqueue");



    return 0;
}
```

Please suggest me what should i do.
And please tell me what exchange type i have to give

I am attaching here my debug screenshot .
![1](https://user-images.githubusercontent.com/9655086/32726172-0088491a-c89e-11e7-812f-24a485e02cb8.png)
![2](https://user-images.githubusercontent.com/9655086/32726183-0aeb43bc-c89e-11e7-8c2f-1734a38de178.png)

![3](https://user-images.githubusercontent.com/9655086/32726182-0a9ebda8-c89e-11e7-929a-51d57c9eabd2.png)

Please suggest me what should i do.
Thanks in advance

**Comment by alanxz** [comment]
Its hard to tell from screenshots - what exception is being thrown? 

All the exceptions thrown by SimpleAmqpClient inherit from std::runtime_error.  Calling the [`what()`](http://en.cppreference.com/w/cpp/error/exception/what) method on the thrown error should give you some information about what is going on.

**Comment by vikaschamp** [comment]
Hello Alanxz,

I am getting error while running my program;--->


**# Program: ...rojects\ConsoleApplication7\x64\Debug\ConsoleApplication7.exe
File: minkernel\crts\ucrt\src\appcrt\heap\debug_heap.cpp
Line: 996

Expression: __acrt_first_block == header

For information on how your program can cause an assertion
failure, see the Visual C++ documentation on asserts.

(Press Retry to debug the application)**

**Comment by vikaschamp** [comment]
Closes #173

**Comment by alanxz** [comment]
I assume you intended on closing this issue.  Re-open if this isn't the case.
